swagger1/src/main/java/com/cerner/beadledom/swagger/SwaggerUiResource.java

```diff
@@ -66,4 +71,40 @@ public Response getSwaggerUiAsset(@PathParam("asset") String assetPath) {
     InputStream stream = getClass().getResourceAsStream("ui-dist/" + assetPath);
     return Response.ok(stream).build();
   }
+
+  /**
+   * Checks if the request was made from a secure context or not.
+   * @param securityContext The {@link SecurityContext} object that could validate security of the request.
+   * @param httpHeaders The {@link HttpHeaders} object containing the request headers.
+   * @return true if the request was made from a secure context, false otherwise.
+   */
+  private boolean isRequestSecure(SecurityContext securityContext, HttpHeaders httpHeaders) {
+    return securityContext.isSecure()
+        || secureForwardedHeader(httpHeaders)
+        || secureXForwardedProtoHeader(httpHeaders);
+  }
+
+  /**
+   * Checks the Forwarded header for a protocol value of https.
+   * @param httpHeaders The {@link HttpHeaders} object containing the request headers.
+   * @return true if the `Forwarded` is present and contains https and false otherwise.
+   */
+  private boolean secureForwardedHeader(HttpHeaders httpHeaders) {
+    String forwardedHeader = httpHeaders.getRequestHeaders().getFirst("Forwarded");
+    Pattern forwardedPairs = Pattern.compile("proto=(?<protocolValue>[^;]*)(;|\\z)");
+
+    return forwardedHeader != null
+        && forwardedPairs.matcher(forwardedHeader).group("protocolValue").contains("https");
+  }
+
+  /**
+   * Checks the X-Forwarded-Proto header for a value of https.
+   * @param httpHeaders The {@link HttpHeaders} object containing the request headers.
+   * @return true if the `X-Forwarded-Proto` header exists and false otherwise.
+   */
+  private boolean secureXForwardedProtoHeader(HttpHeaders httpHeaders) {
+    return httpHeaders.getRequestHeaders().getFirst("X-Forwarded-Proto") != null
```

Comment:
I think reversing the equals would remove the need to check for null:

```java
return "https".equals(httpHeaders.getRequestHeaders().getFirst("X-Forwarded-Proto"));
```

Reply:
Yep: https://github.com/cerner/beadledom/pull/137/commits/448ff0c922f16b3a5fed81879c72a7931826231e